topostats/plotting.py

```diff
@@ -139,31 +139,56 @@ def _outfile(self, plot_suffix: str) -> str:
         """
         return f"{self.stat_to_sum}_{plot_suffix}"
 
-    def sns_plot(self) -> None:
+    def sns_plot(self) -> Optional[Tuple[plt.Figure, plt.Axes]]:
         """Plot the distribution of one or more statistics as either histogram, kernel density estimates or both. Uses
-        base Seaborn."""
+        base Seaborn.
+
+        Returns
+        -------
+        Optional[Union[Tuple[plt.Figure, plt.Axes], None]]
+            Tuple of Matplotlib figure and axes if plotting is successful, None otherwise.
+        """
+
+        # Note: Plotting KDEs with Seaborn is not possible if all values are 0. This is because the KDE is calculated
+        # using a Gaussian kernel and if all values are 0, the standard deviation is 0 wich results in a
+        # ZeroDivisionError. To avoid this, we check if all values are 0 and if so, we skip the KDE plot.
```

Comment:
Sorry for not reading this in detail earlier but if we observe a `ZeroDivsionError` then the natural/Pythonic thing to do would be to use `try: ... except:`

This needs to fit within the logic of whether `self.kde` is true or not.

And so...

```python
if self.hist and not self.kde:
    outfile = self._outfile("hist")
    sns.histplot(data=self.melted_data, x="value", bins=self.bins, stat=self.stat, hue=self.hue)
if self.kde and not self.hist:
    outfile = self._outfile("kde")
    try:
        sns.kde.plot(data=self.melted_data, x="value", hue=self.hue)
    except ZeroDivsionError:
        LOGGER.info(f"[plotting] All values for {self.label} are 0. KDE plots cannot be made as there is no variance, skipping."
        return None
if self.hist and self.kde:
    outfile = self._outfile("hist_kde")
    try:
        sns.histplot(
            data=self.melted_data,
            x="value",
            bins=self.bins,
            stat=self.stat,
            hue=self.hue,
            kde=True,
            kde_kws={"cut": self.cut},
        )
    except ZeroDivisionError:
       LOGGER.info(f"[plotting] All values for {self.label} are 0. KDE plots cannot be made as there is no variance, skipping."
       return None
```

...is one way of doing that. I realise its perhaps not _that_ efficient as its possible to determine if the variance is zero before trying to plot, but it capture instances where all values are the _same_ rather than just zero.